src/AgentCommerce/HoneyWebhookService.php

```diff
@@ -0,0 +1,192 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Swag\PayPal\AgentCommerce;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\ClientException;
+use Lcobucci\JWT\Encoding\ChainedFormatter;
+use Lcobucci\JWT\Encoding\JoseEncoder;
+use Lcobucci\JWT\Signer\Hmac\Sha256;
+use Lcobucci\JWT\Signer\Key\InMemory;
+use Lcobucci\JWT\Token\Builder;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamInterface;
+use Psr\Log\LoggerInterface;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\Country\CountryEntity;
+use Shopware\Core\System\SalesChannel\SalesChannelCollection;
+use Shopware\Core\System\SalesChannel\SalesChannelEntity;
+use Shopware\Core\System\SystemConfig\SystemConfigService;
+use Swag\PayPal\AgentCommerce\Exception\HoneyWebhookExceptions;
+use Swag\PayPal\Setting\Service\CredentialsUtil;
+use Swag\PayPal\Setting\Settings;
+use Swag\PayPal\SwagPayPal;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class HoneyWebhookService
+{
+    protected ClientInterface $client;
+
+    /**
+     * @param EntityRepository<SalesChannelCollection> $salesChannelRepository
+     */
+    public function __construct(
+        private readonly EntityRepository $salesChannelRepository,
+        private readonly CredentialsUtil $credentialsUtil,
+        private readonly RouterInterface $router,
+        private readonly SystemConfigService $systemConfigService,
+        private readonly LoggerInterface $logger,
+    ) {
+        $this->client = new Client([
```

Comment:
you can and properly should inject the client via symfony like it was done [here](https://github.com/shopware/shopware/blob/9d8ddd385cc031e44d8f6c7aef6a366b847fca04/src/Core/Framework/DependencyInjection/app.xml#L332)